lib/instance_billing_flavor_check/utils.py

```diff
@@ -0,0 +1,142 @@
+    else:
+        logger.error("Could not read file %s", REGION_SRV_CLIENT_CONFIG_PATH)
+    return None
+
+
+
+def get_metadata():
+    command = get_command()
+    if not command:
+        return None
+
+    command = command.split(' ')
+    result = Command.run(command)
+    if result.returncode == 0:
+        return result.output
+    else:
+        logger.error(
+            "Could not fetch the metadata after running the command '%s': '%s' with status '%s'",
+            ' '.join(command),
+            result.error,
+            result.returncode
+        )
+        return None
+
+def get_identifier():
+    try:
+        with open(ETC_OS_RELEASE_PATH) as stream:
+            csv_reader = csv.reader(stream, delimiter="=")
+            os_release = dict(csv_reader)
+    except FileNotFoundError:
+        logger.error("Could not open '%s' file", ETC_RELEASE_PATH)
+        syst.exit(1)
+
+    return os_release.get('NAME')
+
+def get_rmt_ip_addr():
+    """ Return the RMT update server IP the instance is registered to."""
+    try:
+        with open(ETC_HOSTS_PATH) as etc_hosts:
+            etc_hosts_lines = etc_hosts.readlines()
+    except FileNotFoundError:
+        logger.error("Could not open '%s' file", ETC_HOSTS_PATH)
+        return
+
+    for etc_hosts_line in etc_hosts_lines:
+        if 'susecloud.net' in etc_hosts_line:
+            return etc_hosts_line.split('\t')[0]
+
+
+def make_request(rmt_ip_addr, metadata, identifier):
+    instance_check_url = f"https://{rmt_ip_addr}/api/instance/check"
+    requests.packages.urllib3.disable_warnings(
+        requests.packages.urllib3.exceptions.InsecureRequestWarning
+    )
+    message = None
+    try:
+        response = requests.get(
+            instance_check_url,
+            timeout=2,
+            verify=False,
+            metadata=metadata,
+            identifer=identifier
+        )
+    except requests.exceptions.HTTPError as err:
+        message = f"Http Error:{err}"
+    except requests.exceptions.ConnectionError as err:
+        message = f"Error Connecting:{err}"
+    except requests.exceptions.Timeout as err:
+        message = f"Timeout Error:{err}"
+    except requests.exceptions.RequestException as err:
+        message = f"Unexpected error:{err}"
+
+    if message:
+        logger.error(message)
+        return
+    if response.status_code == 200:
+        result = response.json()
+        logger.debug(result)
+        return result.get('flavor')
+
+    logger.error(
+        'Request to check if instance is PAYG/BYOS failed: %s', response.reason
+    )
+
+def check_payg_byos():
+    """Return 'PAYG' OR 'BYOS'."""
+    metadata = get_metadata()
+    identifier = get_identifier()
+    if not metadata or not identifier:
+        return
+
+    rmt_ip_addr = get_rmt_ip_addr()
+    if not rmt_ip_addr:
+        logger.warning('Instance can be either BYOS or PAYG and not registered')
+
```

Comment:
There's a logic error here. if we do not find an IP address, we log a warning but still try to make an https call, but we have no address.

Given the error conditions we may encounter, good that these are being considered, maybe it is worth having a conversation with the SUMa team to either define exit codes, for example

10 - PAYG
11- BYOS
12 - Unknown

or return human friendly strings.

Reply:
Updated to print "PAYG", "BYOS" or "Some error" strings + return exit codes based on SUMa feedback